app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java

```diff
@@ -272,13 +284,81 @@ public void onClick(View v) {
     }
 
     public void goToIndividualChat(View view) {
-        Intent intent = new Intent(ActivityDescriptionActivity.this, ChatActivity.class);
-        String activityChatId = act.getActivityId() + " - chatId";
-        intent.putExtra("ACTIVITY_CHAT_ID", activityChatId);
-        String activityTitle = act.getTitle();
-        intent.putExtra("ACTIVITY_TITLE", activityTitle);
-        startActivity(intent);
+        // Allowing the access to the chat only if the user is registered to the activity
+        if (act.getParticipantId().contains(userId)) {
+            Intent intent = new Intent(ActivityDescriptionActivity.this, ChatActivity.class);
+            String activityChatId = act.getActivityId() + " - chatId";
+            intent.putExtra("ACTIVITY_CHAT_ID", activityChatId);
+            String activityTitle = act.getTitle();
+            intent.putExtra("ACTIVITY_TITLE", activityTitle);
+            startActivity(intent);
+        } else {
+            Toast.makeText(ActivityDescriptionActivity.this, "Please register if you want to access the chat!", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private void getOrganizerName() {
+        if (act != null) {
+            organizerId = act.getOrganizerId();
```

Comment:
What is the use of getting the organizer Id if the view is his username ?

Reply:
Because the organiser ID (as well as for instance the user ID) is the key to access the organiser (respectively the user) data on Firebase Firestore. More precisely, the Firebase Firestore document is referenced through the organiser (respectively the user) ID:
DocumentReference docRef = fStore.collection("users").document(organizerId);